### PR TITLE
feat: Prevent bruteforce at login

### PR DESCRIPTION
**Pull request by orronai**
- Added a limiter per ip address in login stage  - 5 per minute, 50 per hour

### CODE REVIEW

File: lms/lmsweb/views.py [hunk 1]
```diff
@@ -104,7 +105,19 @@ def get_next_url(url_next_param: Optional[str]):
...
+    log.info(f'IP Address: {get_remote_address()}: {e}')
+    return make_response(
+        jsonify(error='ratelimit exceeded %s' % e.description), 429,
+    )
+
+
 @webapp.route('/login', methods=['GET', 'POST'])
+@limiter.limit(
+    '5/minute;50/hour',
+    deduct_when=lambda response: response.status_code != 200,
```
Comment (yammesicka) [inline lms/lmsweb/views.py:119]: I think we always return 200, even when the login failed.

File: lms/tests/conftest.py [hunk 2]
```diff
@@ -60,6 +60,7 @@ def webapp_configurations():
     webapp.secret_key = ''.join(
         random.choices(string.ascii_letters + string.digits, k=64),
     )
+    limiter.enabled = False
```
Comment (yammesicka) [inline lms/tests/conftest.py:63]: How about adding a test that on setup sets `limiter.enabled` to `True`, and in the end sets the teardown to `False`?

We must check that the limiter actually works.

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: lms/lmsweb/views.py (hunk 1 above)
Comment (orronai) [reply lms/lmsweb/views.py:119]: Tried it and it works in this way

File: lms/tests/conftest.py (hunk 2 above)
Comment (orronai) [reply lms/tests/conftest.py:63]: How about enabling it in a test, testing it and then disabling it again?

Comment (yammesicka) [reply lms/tests/conftest.py:63]: Yes, good idea

File: lms/lmsweb/views.py (hunk 1 above)
Comment (yammesicka) [reply lms/lmsweb/views.py:119]: You've checked that many refreshes to the login pages doesn't block the user?

Comment (orronai) [reply lms/lmsweb/views.py:119]: You're right.
And yes, I checked it

File: lms/tests/test_flask_limiter.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+from lms.tests import conftest
+from lms.lmsdb.models import User
+from lms.lmsweb import webapp
+
+
+class TestLimiter:
```
Comment (yammesicka) [inline lms/tests/test_flask_limiter.py:6]: Great changes.

We only need to add a test that regular login (with a single try) is working correctly.

Comment (yammesicka) [reply lms/tests/test_flask_limiter.py:6]: Also: add a test that when trying to `GET` the page `/login`, it doesn't apply the rate limit even after `LIMITS_PER_MINUTE` times.

**Review by yammesicka — CHANGES_REQUESTED** [review]

**Review by yammesicka — APPROVED** [review]

**Pull request merged**